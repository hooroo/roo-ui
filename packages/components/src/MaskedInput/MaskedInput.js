import TextMask from 'react-text-mask';

import { Input } from '..';

const MaskedInput = Input.withComponent(TextMask);
MaskedInput.displayName = 'MaskedInput';

MaskedInput.time = MaskedInput.extend``;
MaskedInput.time.displayName = 'MaskedInput.time';
MaskedInput.time.defaultProps = {
  ...MaskedInput.defaultProps,
  keepCharPositions: true,
  placeholderChar: '\u2001',
  mask: [/[0-2]/, /[0-9]/, ':', /[0-6]/, /[0-9]/, ' AEST'],
  placeholder: '00:00 AEST',
};

export default MaskedInput;
