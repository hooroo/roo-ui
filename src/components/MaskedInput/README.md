# Masked input

<!-- STORY -->

## Installation

```shell
$ yarn add roo-ui
```

## Example

```js
import { MaskedInput } from 'roo-ui/components';

export default (
  <Fragment>
    <MaskedInput placeholder="Enter postcode" mask={[/\d/, /\d/, /\d/, /\d/]} />
    <MaskedInput.time />
  </Fragment>
);
```

## Properties

Refer to [React Input Mask](https://github.com/text-mask/text-mask/tree/master/react#readme).

## Customization

This component can be customized with props supported by [`<Input />`](../Input/README.md).
