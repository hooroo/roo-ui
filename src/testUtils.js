import React from 'react';
import { mount, shallow } from 'enzyme';

export const mountWithTheme = (tree, theme) => {
  const themedTree = React.cloneElement(tree, { theme });
  return mount(themedTree);
};

export const shallowWithTheme = (tree, theme) => {
  const themedTree = React.cloneElement(tree, { theme });
  return shallow(themedTree);
};
