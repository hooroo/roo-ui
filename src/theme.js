import { rem } from 'polished';
import range from 'lodash/range';

import * as icons from './defaultIcons';

const namedColors = {
  white: '#FFFFFF',
  brightSun: '#FBCB3B',
  red: '#E00E00',
  maroon: '#870000',
  blue: '#8DE2E0',
  lightBlue: '#E7F7F7',
  orange: '#ED710B',
  lightOrange: '#FCEBCD',
  green: '#35A509',
  lightGreen: '#DEF2DE',
};

const brand = {
  primary: '#E40000',
  secondary: '#8DE2E0',
  tertiary: '#D20000',
  quaternary: '#F9F3E9',
};

const greys = {
  charcoal: '#323232',
  steel: '#666666',
  alto: '#DADADA',
  porcelain: '#F4F5F6',
  dusty: '#999999',
  mineShaft: '#222222',
};

const ui = {
  error: namedColors.orange,
  errorBackground: namedColors.lightOrange,
  info: namedColors.blue,
  infoBackground: namedColors.lightBlue,
  link: namedColors.red,
  linkHover: namedColors.maroon,
  success: namedColors.green,
  successBackground: namedColors.lightGreen,
};

const colors = {
  brand,
  greys,
  ui,
  ...namedColors,
};

const alertStyles = {
  default: { bg: 'greys.porcelain' },
  error: {
    bg: 'ui.errorBackground',
    icon: {
      name: 'warning',
      color: 'ui.error',
    },
  },
  info: {
    bg: 'ui.infoBackground',
    icon: {
      name: 'info',
      color: 'greys.charcoal',
    },
  },
  success: {
    bg: 'ui.successBackground',
    icon: {
      name: 'checkCircle',
      color: 'ui.success',
    },
  },
};

const borders = [0, '1px solid', '2px solid', '3px solid'];

const breakpoints = ['40rem', '52rem', '64rem'];

const buttons = {
  default: {
    color: 'white',
    backgroundColor: 'greys.charcoal',
  },
  primary: {
    color: 'white',
    backgroundColor: 'brand.primary',
  },
};

const fontFamily = 'Ciutadella, sans-serif';

const fontSizes = {
  xs: rem('13px'),
  sm: rem('15px'),
  base: rem('16px'),
  lg: rem('18px'),
  xl: rem('22px'),
  '2xl': rem('26px'),
  '3xl': rem('36px'),
  '4xl': rem('46px'),
};

const fontWeights = {
  normal: 500,
  bold: 600,
};

const letterSpacings = {
  normal: '0',
  wide: '0.125em',
};

const lineHeights = {
  tight: 1,
  normal: 1.35,
  loose: 1.5,
};

const maxWidths = {
  default: rem('1280px'),
};

const mediaQueries = breakpoints.map(
  n => `@media screen and (min-width: ${n})`,
);

const opacity = {
  disabled: 0.7,
};

const radii = {
  none: 0,
  default: '4px',
  rounded: '1000px',
};

const shadows = {
  none: 0,
  default: '0 1px 1px 0 rgba(0, 0, 0, 0.1)',
  heavy: '0 2px 2px 0 rgba(0, 0, 0, 0.1)',
  focus: `0 0 2px 2px ${colors.greys.dusty};`,
};

const space = range(0, 129, 4).map(i => rem(`${i}px`));

const textStyles = {
  caps: {
    textTransform: 'uppercase',
    fontWeight: fontWeights.bold,
  },

  h1: {
    fontSize: fontSizes['3xl'],
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.normal,
  },
  h2: {
    fontSize: fontSizes['2xl'],
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.normal,
  },
  h3: {
    fontSize: fontSizes.xl,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h4: {
    fontSize: fontSizes.lg,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h5: {
    fontSize: fontSizes.base,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  h6: {
    fontSize: fontSizes.sm,
    fontWeight: fontWeights.bold,
    lineHeight: lineHeights.loose,
  },
  label: {
    fontSize: fontSizes.sm,
    lineHeight: lineHeights.loose,
    fontWeight: fontWeights.bold,
    color: colors.greys.charcoal,
  },
  text: {
    fontSize: fontSizes.base,
    lineHeight: lineHeights.loose,
    fontWeight: fontWeights.normal,
  },
};

const transitions = {
  default: '200ms ease-in',
};

export default {
  alertStyles,
  borders,
  breakpoints,
  buttons,
  colors,
  fontFamily,
  fontSizes,
  fontWeights,
  icons,
  letterSpacings,
  lineHeights,
  maxWidths,
  mediaQueries,
  opacity,
  radii,
  shadows,
  space,
  textStyles,
  transitions,
};
