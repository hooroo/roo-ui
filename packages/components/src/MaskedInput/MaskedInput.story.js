import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';
import { boolean } from '@storybook/addon-knobs/react';

import MaskedInput from '.';
import README from './README.md';

storiesOf('Components|MaskedInput', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <MaskedInput
      placeholder="Enter postcode"
      mask={[/\d/, /\d/, /\d/, /\d/]}
      underline={boolean('Underline', false)}
    />
  ))
  .add('time', () => (
    <MaskedInput.time />
  ));
