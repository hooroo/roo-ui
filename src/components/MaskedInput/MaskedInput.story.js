import React from 'react';
import { withDocs } from 'storybook-readme';

import MaskedInput from './';
import README from './README.md';

export default {
  title: 'Components/MaskedInput',
  decorators: [withDocs(README)],
};

export const Default = () => (
  <MaskedInput placeholder="Enter postcode" mask={[/\d/, /\d/, /\d/, /\d/]} />
);

Default.story = {
  name: 'default',
};

export const Time = () => <MaskedInput.time />;

Time.story = {
  name: 'time',
};
