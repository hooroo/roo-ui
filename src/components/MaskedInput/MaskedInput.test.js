import React from 'react';
import theme from 'theme';
import { shallowWithTheme } from 'testUtils';
import { axe } from 'jest-axe';

import MaskedInput from '.';

describe('<MaskedInput />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWithTheme(
      <label htmlFor="postcode">
        Enter postcode
        <MaskedInput
          id="postcode"
          placeholder="Enter postcode"
          mask={[/\d/, /\d/, /\d/, /\d/]}
        />
      </label>,
      theme,
    );
  });

  it('renders correctly', () => {
    expect(wrapper.find('MaskedInput').dive()).toMatchSnapshot();
  });

  it('has no accessibility errors', async () => {
    expect(await axe(wrapper.html())).toHaveNoViolations();
  });

  describe('<MaskedInput.time />', () => {
    beforeEach(() => {
      wrapper = shallowWithTheme(<MaskedInput.time />, theme);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
