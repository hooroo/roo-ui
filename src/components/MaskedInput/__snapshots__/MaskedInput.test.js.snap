// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaskedInput /> <MaskedInput.time /> renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<MaskedInput /> renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
